repose-aggregator/components/filters/regex-rbac-filter/src/main/scala/org/openrepose/filters/regexrbac/RegexRbacFilter.scala

```diff
@@ -0,0 +1,203 @@
+      val (relevantRoles, noMatchingRoles) = matchedMethods.foldLeft(Set.empty[String], List.empty[String]) {
+        case ((matches, noMatch), resource) =>
+          val intersection = resource.roles.intersect(requestRoles)
+          if (intersection.nonEmpty) {
+            (intersection ++ matches, noMatch)
+          } else {
+            (matches, resource.path.string :: noMatch)
+          }
+      }
+      noMatchingRoles match {
+        case Nil => Success(relevantRoles)
+        case _ => Failure(NoMatchingRolesException(noMatchingRoles))
+      }
+    }
+
+    def sendError(message: String, statusCode: Int): Unit = {
+      val status = if (configuration.isMaskRaxRoles403 &&
+        (statusCode == SC_METHOD_NOT_ALLOWED || statusCode == SC_FORBIDDEN)) {
+        logger.debug(s"Masking $statusCode with $SC_NOT_FOUND")
+        SC_NOT_FOUND
+      } else {
+        statusCode
+      }
+
+      Option(configuration.getDelegating) match {
+        case Some(delegating) =>
+          logger.debug(s"Delegating with status $status caused by: {}", message)
+          val delegationHeaders = buildDelegationHeaders(
+            status,
+            Option(delegating.getComponentName).getOrElse("regex-rbac"),
+            s"Failed in the RegEx RBAC filter due to $message",
+            delegating.getQuality)
+          delegationHeaders foreach { case (key, values) =>
+            values foreach { value =>
+              httpRequestWrapper.addHeader(key, value)
+            }
+          }
+          chain.doFilter(httpRequestWrapper, httpResponse)
+        case None =>
+          logger.debug(s"Rejecting with status $status caused by: $message")
+          httpResponse.sendError(status)
+      }
+    }
+
+    processPaths()
+      .flatMap(processMethods)
+      .flatMap(processRoles) match {
+      case Success(relevantRoles) =>
+        httpRequestWrapper.addHeader(XRelevantRolesHeader, relevantRoles.mkString(", "))
+        chain.doFilter(httpRequestWrapper, httpResponse)
+      case Failure(_: NoMatchingPathsException) =>
+        sendError("No Matching Paths", SC_NOT_FOUND)
+      case Failure(_: NoMatchingMethodsException) =>
+        sendError("No Matching Methods", SC_METHOD_NOT_ALLOWED)
+      case Failure(e: NoMatchingRolesException) =>
+        e.paths foreach { path: String =>
+          logger.debug("No Matching Roles: {}", path)
+        }
+        sendError("Non-Matching Roles", SC_FORBIDDEN)
+      case _ =>
+        sendError("Unknown failure", SC_INTERNAL_SERVER_ERROR)
+    }
+  }
+
+  override def doConfigurationUpdated(newConfigurationObject: RegexRbacConfig): Unit = {
+
+    def parseLine(line: String): Option[Resource] = {
+      val values = line.trim.split("\\s+")
+      values.length match {
+        case 3 =>
+          Some(Resource(
+            stringToRegexString(values(0)),
+            Try(values(1).split(',').toSet[String].map(_.trim) map (_.toUpperCase)).getOrElse(Set.empty),
```

Comment:
One of the maps on this line is called without using dot notation, which seems inconsistent with other usages.

Reply:
Yep. Completely agree with that.
Updated.